internal/backend/fs.go

```diff
@@ -217,28 +227,30 @@ func (s *storageFS) ListObjects(bucketName string, prefix string, versions bool)
 	s.mtx.RLock()
 	defer s.mtx.RUnlock()
 
-	infos, err := os.ReadDir(filepath.Join(s.rootDir, url.PathEscape(bucketName)))
-	if err != nil {
-		return nil, err
-	}
 	objects := []ObjectAttrs{}
-	for _, info := range infos {
+	// TODO: WalkDir more efficient?
+	bucketPath := filepath.Join(s.rootDir, url.PathEscape(bucketName))
+	if err := filepath.Walk(bucketPath, func(path string, info fs.FileInfo, _ error) error {
+		objName := strings.TrimPrefix(path, bucketPath+string(filepath.Separator))
```

Comment:
Use `filepath.Rel` instead?
```suggestion
		objName := filepath.Rel(bucketPath, path)
```